Step-by-step-Tutorials/Custom-Storage-Providers.md

```diff
@@ -167,9 +168,9 @@ The format decisions have already been made, so coding up the `WriteStateAsync`
 
 
 ``` csharp
-public async Task WriteStateAsync(string grainType, Orleans.GrainReference grainRef, IGrainState grainState)
+public async Task WriteStateAsync(string grainType, GrainReference grainRef, IGrainState grainState)
 {
-    Dictionary<string, object> dataValues = grainState.AsDictionary();
+    IDictionary<string, object> dataValues = ((GrainState)grainState.State).AsDictionary();
     JavaScriptSerializer serializer = new JavaScriptSerializer();
```

Comment:
@jdom, thanks for giving me the pointer to look into the AzureBlobStorage. Then, we can leverage a minimal usage of JsonConvert class for this example, similar to the one in AzureBlobStorage.
However, about ETag property, I'm not sure whether we should add the checks to the code of this tutorial (which probably is one of the first steps for someone that ones to try out Orleans), or we should just mention the behavior and give a pointer to a more involved description (possibly along with examples).